File: js/LongestPalindrome.js
var longestPalindrome = function(s) {
    let dict = {}
    for(let i = 0; i < s.length; i++){
        if(!dict[s[i]] ){
            dict[s[i]] = 0;
        }
        dict[s[i]] ++;
    }
    let rst = 0;
    let flag = false;
   Object.keys(dict).forEach(key => {
        value = dict[key];
        rst += Math.floor(value /2) * 2;
        if(value % 2 == 1){
            flag = true;
        }

   })
    return flag? rst + 1: rst;
};

console.log(longestPalindrome("abccccdd"));

File: python/LongestPalindrome.py
__author__ = ' Zhen Wang'

import collections

class LongestPalindrome:
    def longestPalindrome(self, s: str) -> int:
        rst = 0
        flag = False
        d = collections.Counter(s)
        for  key in d:
            rst += d[key]//2 * 2
            if d[key] %2 == 1:
                flag = True
        return rst + 1 if flag else rst


if __name__ == "__main__":
    s = "abccccdd"
    lp = LongestPalindrome()
    rst = lp.longestPalindrome(s)
    print(rst)
